<a href="https://colab.research.google.com/github/Jaomiew/data_viz_2022/blob/main/Final_Exam_DataViz.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [1]:
import pandas as pd     # ใช้ทำงานกับ ตาราง

In [2]:
from google.colab import drive
drive.mount('/content/drive')

Mounted at /content/drive


In [3]:
%cd /content/drive/MyDrive/Final Exam Dataviz

/content/drive/MyDrive/Final Exam Dataviz


# **alldb**

In [4]:
alldb = pd.read_csv('alldb.csv')
print(alldb.shape)
alldb.head()

(40035, 14)


Unnamed: 0           user                                    RowKey care  \
0           0           test                                       NaN  NaN   
1           1          04857      136FD222-FD9F-4E54-AE1F-7282A276B085  NaN   
2           2            NaN  1E6D3D62-4C8A-4F8E-90D1-0DC5BC2752DF.png  NaN   
3           3  3450101576642      2D331FB0-CBA2-4586-A5B0-CB3CF44DD9FE  NaN   
4           4  3450101576642      45197AF7-7CB4-470C-9ABB-37C092BD6B21  NaN   

         date   ov       phone   site      time            urine  pid  \
0         NaN  NaN         NaN    NaN       NaN              NaN  NaN   
1  2022-02-24  NaN  iPhone10,4  05310  10:50:32              NaN  NaN   
2         NaN  NaN         NaN    NaN       NaN              NaN  NaN   
3  2022-02-23    0  iPhone13,4  05087  09:59:47  ปกติ (เหลืองใส)  NaN   
4  2022-02-23  NaN  iPhone13,4  05087  09:44:06  ปกติ (เหลืองใส)  NaN   

  PartitionKey                  title                    age  
0          NaN  data is not completed  data is not completed  
1          NaN  data is not completed  data is not completed  
2          NaN  data is not completed  data is not completed  
3          NaN  data is not completed  data is not completed  
4          NaN  data is not completed  data is not completed

In [5]:
alldb = alldb.drop(columns=['Unnamed: 0','care','pid'])     # ลบ คอลัมน์ที่ไม่ต้องการออก
print(alldb.shape)
alldb.head()

(40035, 11)


user                                    RowKey        date   ov  \
0           test                                       NaN         NaN  NaN   
1          04857      136FD222-FD9F-4E54-AE1F-7282A276B085  2022-02-24  NaN   
2            NaN  1E6D3D62-4C8A-4F8E-90D1-0DC5BC2752DF.png         NaN  NaN   
3  3450101576642      2D331FB0-CBA2-4586-A5B0-CB3CF44DD9FE  2022-02-23    0   
4  3450101576642      45197AF7-7CB4-470C-9ABB-37C092BD6B21  2022-02-23  NaN   

        phone   site      time            urine PartitionKey  \
0         NaN    NaN       NaN              NaN          NaN   
1  iPhone10,4  05310  10:50:32              NaN          NaN   
2         NaN    NaN       NaN              NaN          NaN   
3  iPhone13,4  05087  09:59:47  ปกติ (เหลืองใส)          NaN   
4  iPhone13,4  05087  09:44:06  ปกติ (เหลืองใส)          NaN   

                   title                    age  
0  data is not completed  data is not completed  
1  data is not completed  data is not completed  
2  data is not completed  data is not completed  
3  data is not completed  data is not completed  
4  data is not completed  data is not completed

In [6]:
alldb.isnull().sum()

user            762
RowKey            1
date            554
ov              364
phone             3
site             99
time            557
urine           253
PartitionKey    185
title           444
age             542
dtype: int64

## **C: user**

**user เป็นตัวเลข 5 หลัก (ตัวเลขเท่านั้น)**

In [7]:
alldb = alldb[(alldb['user'].astype(str).str.isdigit()) & (alldb['user'].str.len() == 5)]      # เลือกมาเฉพาะ คอลัมน์ user ที่เป็นตัวเลข & มีตัวเลข เท่ากับ 5 ตัว
print(alldb.shape)                                                                             # .str.isdigit ใช้ตรวจสอบว่าอักขระทั้งหมดเป็นตัวเลขหรือไม่
alldb.head()

(38508, 11)


user                                RowKey        date   ov       phone  \
1   04857  136FD222-FD9F-4E54-AE1F-7282A276B085  2022-02-24  NaN  iPhone10,4   
5   08777  472EED3E-E2A7-4551-9249-D87747C68AFD  2022-02-25  NaN  iPhone13,2   
7   04857  73E96EE6-7660-4204-9914-92112603C514  2022-02-24  NaN  iPhone10,4   
11  08777  C23A546B-EB6E-4DB4-93FA-2F9A7A85AEE9  2022-02-25  NaN  iPhone13,2   
12  11111  C42C6F97-4388-4C03-A0AB-84F21FAED2C2  2022-02-27  NaN  iPhone13,3   

     site      time urine PartitionKey                  title  \
1   05310  10:50:32   NaN          NaN  data is not completed   
5   05248  10:14:46   NaN          NaN  data is not completed   
7   05310  11:14:21   NaN          NaN  data is not completed   
11  05248  10:14:32   NaN          NaN  data is not completed   
12  22222  17:39:35   NaN          NaN  data is not completed   

                      age  
1   data is not completed  
5   data is not completed  
7   data is not completed  
11  data is not completed  
12  data is not completed

**ที่มาโค้ด**



ref1: https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.Series.str.isdigit.html

ref2: https://stackoverflow.com/questions/24883503/pandas-select-only-numeric-or-integer-field-from-dataframe


In [8]:
alldb.isnull().sum()      # ดูค่าว่าง

user              0
RowKey            0
date            302
ov              221
phone             0
site              2
time            303
urine           126
PartitionKey     25
title           413
age             511
dtype: int64

## **C: site**

**site เป็นตัวเลข 5 หลัก (ตัวเลขเท่านั้น)**

In [9]:
alldb = alldb[(alldb['site'].astype(str).str.isdigit()) & (alldb['site'].str.len() == 5)]      # เลือกมาเฉพาะ คอลัมน์ site ที่เป็นตัวเลข & มีตัวเลข เท่ากับ 5 ตัว
print(alldb.shape)                                                                             # .str.isdigit ใช้ตรวจสอบว่าอักขระทั้งหมดเป็นตัวเลขหรือไม่
alldb.head()

(38493, 11)


user                                RowKey        date   ov       phone  \
1   04857  136FD222-FD9F-4E54-AE1F-7282A276B085  2022-02-24  NaN  iPhone10,4   
5   08777  472EED3E-E2A7-4551-9249-D87747C68AFD  2022-02-25  NaN  iPhone13,2   
7   04857  73E96EE6-7660-4204-9914-92112603C514  2022-02-24  NaN  iPhone10,4   
11  08777  C23A546B-EB6E-4DB4-93FA-2F9A7A85AEE9  2022-02-25  NaN  iPhone13,2   
12  11111  C42C6F97-4388-4C03-A0AB-84F21FAED2C2  2022-02-27  NaN  iPhone13,3   

     site      time urine PartitionKey                  title  \
1   05310  10:50:32   NaN          NaN  data is not completed   
5   05248  10:14:46   NaN          NaN  data is not completed   
7   05310  11:14:21   NaN          NaN  data is not completed   
11  05248  10:14:32   NaN          NaN  data is not completed   
12  22222  17:39:35   NaN          NaN  data is not completed   

                      age  
1   data is not completed  
5   data is not completed  
7   data is not completed  
11  data is not completed  
12  data is not completed

In [10]:
alldb.isnull().sum()      # ดูค่าว่าง

user              0
RowKey            0
date            300
ov              219
phone             0
site              0
time            301
urine           124
PartitionKey     20
title           409
age             507
dtype: int64

## **C: PartitionKey**

**PartitionKey เป็นตัวเลข 5 หลัก (ตัวเลขเท่านั้น)**

In [11]:
alldb = alldb[(alldb['PartitionKey'].astype(str).str.isdigit()) & (alldb['PartitionKey'].str.len() == 5)]      # เลือกมาเฉพาะ คอลัมน์ site ที่เป็นตัวเลข & มีตัวเลข เท่ากับ 5 ตัว
print(alldb.shape)                                                                             # .str.isdigit ใช้ตรวจสอบว่าอักขระทั้งหมดเป็นตัวเลขหรือไม่
alldb.head()

(38382, 11)


user                                RowKey        date ov  \
190  06223                        kfpwuaqldj.png  2022-03-23  0   
191  12015  00859DB9-1BD9-4825-B246-3DC25EFD1924  2022-03-30  4   
192  91811  1376ACA2-AFA5-4457-967A-D2FCF908E6A5  2022-03-30  4   
193  95821  1914EE67-CD8F-47EE-A968-2772911AA5DF  2022-03-30  3   
194  13361  1A219EE7-3647-4D7D-A8C0-AE638E69D889  2022-03-29  3   

                 phone   site      time            urine PartitionKey   title  \
190  Xiaomi 21081111RG  13777  16:31:45  ปกติ (เหลืองใส)        00000     NaN   
191         iPhone12,3  06307  14:46:08  ปกติ (เหลืองใส)        00001  นางสาว   
192         iPhone14,3  05328  15:27:45  ปกติ (เหลืองใส)        00001     นาง   
193         iPhone13,4  06302  10:16:46  ปกติ (เหลืองใส)        00001     นาง   
194          iPhone9,4  05059  12:26:38  ปกติ (เหลืองใส)        00001     นาย   

     age  
190  NaN  
191   47  
192   41  
193   55  
194   66

In [12]:
alldb.isnull().sum()      # ดูค่าว่าง

user              0
RowKey            0
date            282
ov              210
phone             0
site              0
time            283
urine           114
PartitionKey      0
title           407
age             505
dtype: int64

## **C: ov**

**ov เป็นตัวเลข มีค่าตั้งแต่ 0-4 (0 คือไม่พบโรค 1-4 คือพบโรคที่ระดับความรุนแรงต่างๆ)**

In [13]:
set(alldb['ov'])

{'0', '1', '2', '3', '4', nan, 'ติดเชื้อ +2 '}

In [14]:
alldb[alldb['ov']=='ติดเชื้อ +2 ']

user                                RowKey        date            ov  \
16931  87374  A838A03E-1327-4720-B689-FD29C78023FD  2022-03-07  ติดเชื้อ +2    

           phone   site      time            urine PartitionKey title age  
16931  iPhone8,2  05237  11:50:47  ปกติ (เหลืองใส)        00187   นาง  77

- จาก คอลัมน์ ov ที่มีค่า nan = 210 และ ค่า 'ติดเชื้อ +2 ' = 1 ค่า ทำการลบ แถวดังกล่าวทิ้ง

In [15]:
alldb = alldb.drop(16931)       # ลบแถว 16931
print(alldb.shape)
alldb.head(3)

(38381, 11)


user                                RowKey        date ov  \
190  06223                        kfpwuaqldj.png  2022-03-23  0   
191  12015  00859DB9-1BD9-4825-B246-3DC25EFD1924  2022-03-30  4   
192  91811  1376ACA2-AFA5-4457-967A-D2FCF908E6A5  2022-03-30  4   

                 phone   site      time            urine PartitionKey   title  \
190  Xiaomi 21081111RG  13777  16:31:45  ปกติ (เหลืองใส)        00000     NaN   
191         iPhone12,3  06307  14:46:08  ปกติ (เหลืองใส)        00001  นางสาว   
192         iPhone14,3  05328  15:27:45  ปกติ (เหลืองใส)        00001     นาง   

     age  
190  NaN  
191   47  
192   41

In [16]:
alldb = alldb.dropna(subset=['ov']) 
print(alldb.shape)
alldb.head(3)

(38171, 11)


user                                RowKey        date ov  \
190  06223                        kfpwuaqldj.png  2022-03-23  0   
191  12015  00859DB9-1BD9-4825-B246-3DC25EFD1924  2022-03-30  4   
192  91811  1376ACA2-AFA5-4457-967A-D2FCF908E6A5  2022-03-30  4   

                 phone   site      time            urine PartitionKey   title  \
190  Xiaomi 21081111RG  13777  16:31:45  ปกติ (เหลืองใส)        00000     NaN   
191         iPhone12,3  06307  14:46:08  ปกติ (เหลืองใส)        00001  นางสาว   
192         iPhone14,3  05328  15:27:45  ปกติ (เหลืองใส)        00001     นาง   

     age  
190  NaN  
191   47  
192   41

In [17]:
set(alldb['ov'])    # ดู set ของข้อมูล

{'0', '1', '2', '3', '4'}

In [18]:
alldb.isnull().sum()      # ดูค่าว่าง

user              0
RowKey            0
date            273
ov                0
phone             0
site              0
time            274
urine             2
PartitionKey      0
title           406
age             503
dtype: int64

## **C: date**

**date เป็น YYYY-MM-DD**

In [19]:
alldb['date'] = pd.to_datetime(alldb['date'],format="%Y-%m-%d")     # เปลี่ยนชนิดข้อมูล และจัดรูปแบบข้อมูล
print(alldb.shape)
alldb.head(3)

(38171, 11)


user                                RowKey       date ov  \
190  06223                        kfpwuaqldj.png 2022-03-23  0   
191  12015  00859DB9-1BD9-4825-B246-3DC25EFD1924 2022-03-30  4   
192  91811  1376ACA2-AFA5-4457-967A-D2FCF908E6A5 2022-03-30  4   

                 phone   site      time            urine PartitionKey   title  \
190  Xiaomi 21081111RG  13777  16:31:45  ปกติ (เหลืองใส)        00000     NaN   
191         iPhone12,3  06307  14:46:08  ปกติ (เหลืองใส)        00001  นางสาว   
192         iPhone14,3  05328  15:27:45  ปกติ (เหลืองใส)        00001     นาง   

     age  
190  NaN  
191   47  
192   41

In [20]:
alldb['date'] = alldb[['date']].fillna(method="ffill")      # เติมค่าว่างของวันที่ด้วย ค่าก่อนหน้า หรือค่าข้างบน
print(alldb.shape)
alldb.head(3)

(38171, 11)


user                                RowKey       date ov  \
190  06223                        kfpwuaqldj.png 2022-03-23  0   
191  12015  00859DB9-1BD9-4825-B246-3DC25EFD1924 2022-03-30  4   
192  91811  1376ACA2-AFA5-4457-967A-D2FCF908E6A5 2022-03-30  4   

                 phone   site      time            urine PartitionKey   title  \
190  Xiaomi 21081111RG  13777  16:31:45  ปกติ (เหลืองใส)        00000     NaN   
191         iPhone12,3  06307  14:46:08  ปกติ (เหลืองใส)        00001  นางสาว   
192         iPhone14,3  05328  15:27:45  ปกติ (เหลืองใส)        00001     นาง   

     age  
190  NaN  
191   47  
192   41

In [21]:
print(alldb.dtypes)
print('*'*50)
alldb.isnull().sum()

user                    object
RowKey                  object
date            datetime64[ns]
ov                      object
phone                   object
site                    object
time                    object
urine                   object
PartitionKey            object
title                   object
age                     object
dtype: object
**************************************************


user              0
RowKey            0
date              0
ov                0
phone             0
site              0
time            274
urine             2
PartitionKey      0
title           406
age             503
dtype: int64

## **C: time**

**time เป็น hh-mm-ss**

In [22]:
alldb['time'] = alldb[['time']].fillna(method="ffill")      # เติมค่าว่างของเวลาด้วย ค่าก่อนหน้า หรือค่าข้างบน
print(alldb.shape)
alldb.head(3)

(38171, 11)


user                                RowKey       date ov  \
190  06223                        kfpwuaqldj.png 2022-03-23  0   
191  12015  00859DB9-1BD9-4825-B246-3DC25EFD1924 2022-03-30  4   
192  91811  1376ACA2-AFA5-4457-967A-D2FCF908E6A5 2022-03-30  4   

                 phone   site      time            urine PartitionKey   title  \
190  Xiaomi 21081111RG  13777  16:31:45  ปกติ (เหลืองใส)        00000     NaN   
191         iPhone12,3  06307  14:46:08  ปกติ (เหลืองใส)        00001  นางสาว   
192         iPhone14,3  05328  15:27:45  ปกติ (เหลืองใส)        00001     นาง   

     age  
190  NaN  
191   47  
192   41

In [23]:
alldb['time'] = pd.to_datetime(alldb['time'],format='%H:%M%:%S',errors='ignore')    # เปลี่ยนชนิดข้อมูล และจัดรูปแบบข้อมูล
print(alldb.shape)
alldb.head(3)

(38171, 11)


user                                RowKey       date ov  \
190  06223                        kfpwuaqldj.png 2022-03-23  0   
191  12015  00859DB9-1BD9-4825-B246-3DC25EFD1924 2022-03-30  4   
192  91811  1376ACA2-AFA5-4457-967A-D2FCF908E6A5 2022-03-30  4   

                 phone   site      time            urine PartitionKey   title  \
190  Xiaomi 21081111RG  13777  16:31:45  ปกติ (เหลืองใส)        00000     NaN   
191         iPhone12,3  06307  14:46:08  ปกติ (เหลืองใส)        00001  นางสาว   
192         iPhone14,3  05328  15:27:45  ปกติ (เหลืองใส)        00001     นาง   

     age  
190  NaN  
191   47  
192   41

In [24]:
alldb.isnull().sum()

user              0
RowKey            0
date              0
ov                0
phone             0
site              0
time              0
urine             2
PartitionKey      0
title           406
age             503
dtype: int64

## **C: urine**

**urine มีค่า String**

In [25]:
alldb['urine'] = alldb[['urine']].fillna(method="ffill")      # เติมค่าว่างของ urine(ปัสสาวะ) ด้วย ค่าก่อนหน้า หรือค่าข้างบน
print(alldb.shape)
alldb.head(3)

(38171, 11)


user                                RowKey       date ov  \
190  06223                        kfpwuaqldj.png 2022-03-23  0   
191  12015  00859DB9-1BD9-4825-B246-3DC25EFD1924 2022-03-30  4   
192  91811  1376ACA2-AFA5-4457-967A-D2FCF908E6A5 2022-03-30  4   

                 phone   site      time            urine PartitionKey   title  \
190  Xiaomi 21081111RG  13777  16:31:45  ปกติ (เหลืองใส)        00000     NaN   
191         iPhone12,3  06307  14:46:08  ปกติ (เหลืองใส)        00001  นางสาว   
192         iPhone14,3  05328  15:27:45  ปกติ (เหลืองใส)        00001     นาง   

     age  
190  NaN  
191   47  
192   41

In [26]:
print(alldb.dtypes)
print('*'*50)
alldb.isnull().sum()

user                    object
RowKey                  object
date            datetime64[ns]
ov                      object
phone                   object
site                    object
time                    object
urine                   object
PartitionKey            object
title                   object
age                     object
dtype: object
**************************************************


user              0
RowKey            0
date              0
ov                0
phone             0
site              0
time              0
urine             0
PartitionKey      0
title           406
age             503
dtype: int64

## **C: title (sex)**

**(เปลี่ยนข้อมูลเป็น เพศ ชาย และ หญิง เพื่อให้นำข้อมูลไปใช้ประโยชน์ได้)**

In [27]:
alldb = alldb.rename(columns={'title':'sex'})  #     เปลี่ยนชื่อ คอลัมน์ title เป็น sex
print(alldb.shape)
alldb.head(3)

(38171, 11)


user                                RowKey       date ov  \
190  06223                        kfpwuaqldj.png 2022-03-23  0   
191  12015  00859DB9-1BD9-4825-B246-3DC25EFD1924 2022-03-30  4   
192  91811  1376ACA2-AFA5-4457-967A-D2FCF908E6A5 2022-03-30  4   

                 phone   site      time            urine PartitionKey     sex  \
190  Xiaomi 21081111RG  13777  16:31:45  ปกติ (เหลืองใส)        00000     NaN   
191         iPhone12,3  06307  14:46:08  ปกติ (เหลืองใส)        00001  นางสาว   
192         iPhone14,3  05328  15:27:45  ปกติ (เหลืองใส)        00001     นาง   

     age  
190  NaN  
191   47  
192   41

In [28]:
set(alldb['sex'])   # ดู set ของข้อมูล

{'  นาง',
 ' นาง',
 ' นางสาว',
 ' นาย',
 'Mrs',
 nan,
 'ชาย',
 'ด.ช.',
 'ด.ญ.',
 'ด.ต.',
 'น',
 'น.ส',
 'น.ส ',
 'น.ส.',
 'นส',
 'นส.',
 'นส. ',
 'นาง',
 'นาง ',
 'นาง.',
 'นางl',
 'นางสาว',
 'นางสาว ',
 'นางสาว                   ',
 'นาน',
 'นาบ',
 'นาย',
 'นาย ',
 'นาย  ',
 'นาว',
 'นาวสาว',
 'นา่ง',
 'น่าย',
 'พจอ.',
 'พระ',
 'พระครุ',
 'พระปลัด',
 'พลฯ',
 'ร.ต.',
 'ร.ต.ต.',
 'ร.ต.ท.',
 'ร.ต.อ.',
 'รต.',
 'ว่าที่ ร.ต.',
 'ว่าที่ ร.ท.',
 'ว่าที่ร้อยตรี',
 'ส.อ.',
 'ืนาง',
 'เด็กชาย',
 'เด็กหญิง'}

In [29]:
# เปลี่ยนให้ข้อมูลเป็น เพศ
alldb['sex'].replace({'  นาง':'หญิง',
                    ' นาง':'หญิง',
                    ' นางสาว':'หญิง',
                    ' นาย':'ชาย',
                    'Mrs':'หญิง',
                    'ชาย':'ชาย',
                    'ด.ช.':'ชาย',
                    'ด.ญ.':'หญิง',
                    'ด.ต.':'ไม่ระบุ',
                    'น':'ไม่ระบุ',
                    'น.ส':'หญิง',
                    'น.ส ':'หญิง',
                    'น.ส.':'หญิง',
                    'นส':'หญิง',
                    'นส.':'หญิง',
                    'นส. ':'หญิง',
                    'นาง':'หญิง',
                    'นาง ':'หญิง',
                    'นาง.':'หญิง',
                    'นางl':'หญิง',
                    'นางสาว':'หญิง',
                    'นางสาว ':'หญิง',
                    'นางสาว                   ':'หญิง',
                    'นาน':'ชาย',
                    'นาบ':'ชาย',
                    'นาย':'ชาย',
                    'นาย ':'ชาย',
                    'นาย  ':'ชาย',
                    'นาว':'หญิง',
                    'นาวสาว':'หญิง',
                    'นา่ง':'หญิง',
                    'น่าย':'ชาย',
                    'พจอ.':'ไม่ระบุ',
                    'พระ':'ชาย',
                    'พระครุ':'ไม่ระบุ',
                    'พระปลัด':'ไม่ระบุ',
                    'พลฯ':'ไม่ระบุ',
                    'ร.ต.':'ไม่ระบุ',
                    'ร.ต.ต.':'ไม่ระบุ',
                    'ร.ต.ท.':'ไม่ระบุ',
                    'ร.ต.อ.':'ไม่ระบุ',
                    'รต.':'ไม่ระบุ',
                    'ว่าที่ ร.ต.':'ไม่ระบุ',
                    'ว่าที่ ร.ท.':'ไม่ระบุ',
                    'ว่าที่ร้อยตรี':'ไม่ระบุ',
                    'ส.อ.':'ไม่ระบุ',
                    'ืนาง':'หญิง',
                    'เด็กชาย':'ชาย',
                    'เด็กหญิง':'หญิง'},inplace=True)

In [30]:
set(alldb['sex'])

{nan, 'ชาย', 'หญิง', 'ไม่ระบุ'}

In [31]:
alldb['sex'] = alldb[['sex']].fillna('ไม่ระบุ')      # เติมค่าว่างของ sex ด้วย ไม่ระบุ
print(alldb.shape)
alldb.head(3)

(38171, 11)


user                                RowKey       date ov  \
190  06223                        kfpwuaqldj.png 2022-03-23  0   
191  12015  00859DB9-1BD9-4825-B246-3DC25EFD1924 2022-03-30  4   
192  91811  1376ACA2-AFA5-4457-967A-D2FCF908E6A5 2022-03-30  4   

                 phone   site      time            urine PartitionKey  \
190  Xiaomi 21081111RG  13777  16:31:45  ปกติ (เหลืองใส)        00000   
191         iPhone12,3  06307  14:46:08  ปกติ (เหลืองใส)        00001   
192         iPhone14,3  05328  15:27:45  ปกติ (เหลืองใส)        00001   

         sex  age  
190  ไม่ระบุ  NaN  
191     หญิง   47  
192     หญิง   41

In [32]:
print(set(alldb['sex']))
print('*'*50)
alldb.isnull().sum()

{'หญิง', 'ชาย', 'ไม่ระบุ'}
**************************************************


user              0
RowKey            0
date              0
ov                0
phone             0
site              0
time              0
urine             0
PartitionKey      0
sex               0
age             503
dtype: int64

## **C: age**

In [33]:
alldb['age'] = alldb['age'].astype(float).astype('Int64')       # แปลงข้อมูล จาก object เป็น int ด้วยการ แปลงจาก object -> float -> int64
print(alldb.dtypes)

user                    object
RowKey                  object
date            datetime64[ns]
ov                      object
phone                   object
site                    object
time                    object
urine                   object
PartitionKey            object
sex                     object
age                      Int64
dtype: object


In [34]:
outlier = alldb[(alldb['age'] < 1) | (alldb['age'] > 150)] # เลือกอายุที่น้อยกว่า 1 หรือ มากว่า 150 ปี
print(outlier.shape)
outlier.head()

(38, 11)


user                                RowKey       date ov  \
908   54538                        gglidedofd.png 2022-03-31  0   
2951  06260  86FCFED2-1EC4-411D-B263-63ECB93DF740 2022-03-30  0   
3224  18331  D1119250-2FE6-43DD-B265-0D09CCEFA6B2 2022-03-30  0   
3657  06260  2227F98F-CEFF-4EB8-93DA-249068E49B0D 2022-03-30  4   
4225  22680  58509197-31BA-4917-B28F-3469578FDC64 2022-03-15  0   

                 phone   site      time            urine PartitionKey   sex  \
908   Samsung SM-A605G  04399  09:40:44  ปกติ (เหลืองใส)        00005  หญิง   
2951        iPhone13,4  06260  09:30:39  ปกติ (เหลืองใส)        00020  หญิง   
3224        iPhone12,1  06255  14:21:10  ปกติ (เหลืองใส)        00022  หญิง   
3657        iPhone13,4  06260  09:51:27  ปกติ (เหลืองใส)        00026  หญิง   
4225         iPhone8,2  04457  10:19:00  ปกติ (เหลืองใส)        00031  หญิง   

       age  
908      0  
2951     0  
3224  -233  
3657     0  
4225   315

In [35]:
alldb[['age']] = alldb[['age']].replace(dict.fromkeys(set(outlier['age']), pd.NA))      # เปลี่ยนค่าผิดปกติเป็นค่าว่าง
print(alldb.shape)
alldb.head(3)

(38171, 11)


user                                RowKey       date ov  \
190  06223                        kfpwuaqldj.png 2022-03-23  0   
191  12015  00859DB9-1BD9-4825-B246-3DC25EFD1924 2022-03-30  4   
192  91811  1376ACA2-AFA5-4457-967A-D2FCF908E6A5 2022-03-30  4   

                 phone   site      time            urine PartitionKey  \
190  Xiaomi 21081111RG  13777  16:31:45  ปกติ (เหลืองใส)        00000   
191         iPhone12,3  06307  14:46:08  ปกติ (เหลืองใส)        00001   
192         iPhone14,3  05328  15:27:45  ปกติ (เหลืองใส)        00001   

         sex   age  
190  ไม่ระบุ  <NA>  
191     หญิง    47  
192     หญิง    41

In [36]:
alldb.isnull().sum()

user              0
RowKey            0
date              0
ov                0
phone             0
site              0
time              0
urine             0
PartitionKey      0
sex               0
age             541
dtype: int64

In [37]:
mean_age = alldb['age'].mean()      # หาอายุเฉลี่ย
mean_age

53.91461599787404

In [38]:
alldb['age'] = alldb['age'].fillna(54)     # แทนค่า nan ของ age ด้วย mean

In [39]:
print(alldb.dtypes)
print('*'*50)
alldb.isnull().sum()

user                    object
RowKey                  object
date            datetime64[ns]
ov                      object
phone                   object
site                    object
time                    object
urine                   object
PartitionKey            object
sex                     object
age                      int64
dtype: object
**************************************************


user            0
RowKey          0
date            0
ov              0
phone           0
site            0
time            0
urine           0
PartitionKey    0
sex             0
age             0
dtype: int64

In [40]:
print(f'เสียข้อมูลไป {40035-38171} แถว')
print(f'คิดเป็น {((40035-38171)*100)/40035} %')

เสียข้อมูลไป 1864 แถว
คิดเป็น 4.655926064693393 %


**เสียข้อมูลไป 1864 แถว**

**คิดเป็น 4.655926064693393 %**

In [41]:
print(alldb.shape)      # ตารางที่ clean เสร็จแล้ว
alldb.head()

(38171, 11)


user                                RowKey       date ov  \
190  06223                        kfpwuaqldj.png 2022-03-23  0   
191  12015  00859DB9-1BD9-4825-B246-3DC25EFD1924 2022-03-30  4   
192  91811  1376ACA2-AFA5-4457-967A-D2FCF908E6A5 2022-03-30  4   
193  95821  1914EE67-CD8F-47EE-A968-2772911AA5DF 2022-03-30  3   
194  13361  1A219EE7-3647-4D7D-A8C0-AE638E69D889 2022-03-29  3   

                 phone   site      time            urine PartitionKey  \
190  Xiaomi 21081111RG  13777  16:31:45  ปกติ (เหลืองใส)        00000   
191         iPhone12,3  06307  14:46:08  ปกติ (เหลืองใส)        00001   
192         iPhone14,3  05328  15:27:45  ปกติ (เหลืองใส)        00001   
193         iPhone13,4  06302  10:16:46  ปกติ (เหลืองใส)        00001   
194          iPhone9,4  05059  12:26:38  ปกติ (เหลืองใส)        00001   

         sex  age  
190  ไม่ระบุ   54  
191     หญิง   47  
192     หญิง   41  
193     หญิง   55  
194      ชาย   66

# **hospital_list**

In [42]:
hospital = pd.read_excel('hospital_list.xlsx')
print(hospital.shape)
hospital.head()

(17103, 16)


hcode                                     name  zone_code  \
0  00001      สำนักงานสาธารณสุขจังหวัดสมุทรปราการ          6   
1  00002          สำนักงานสาธารณสุขจังหวัดนนทบุรี          4   
2  00003         สำนักงานสาธารณสุขจังหวัดปทุมธานี          4   
3  00004  สำนักงานสาธารณสุขจังหวัดพระนครศรีอยุธยา          4   
4  00005          สำนักงานสาธารณสุขจังหวัดอ่างทอง          4   

              zone_name  provincecode         province  amphurcode  \
0    สาขาพื้นที่(ระยอง)          11.0      สมุทรปราการ         1.0   
1  สาขาพื้นที่(สระบุรี)          12.0          นนทบุรี         1.0   
2  สาขาพื้นที่(สระบุรี)          13.0         ปทุมธานี         1.0   
3  สาขาพื้นที่(สระบุรี)          14.0  พระนครศรีอยุธยา         1.0   
4  สาขาพื้นที่(สระบุรี)          15.0          อ่างทอง         1.0   

             amphur  tamboncode     tambon  code4                     name5  \
0  เมืองสมุทรปราการ         1.0     ปากน้ำ    1.0  สำนักงานสาธารณสุขจังหวัด   
1      เมืองนนทบุรี         4.0   บางกระสอ    1.0  สำนักงานสาธารณสุขจังหวัด   
2     เมืองปทุมธานี         1.0    บางปรอก    1.0  สำนักงานสาธารณสุขจังหวัด   
3   พระนครศรีอยุธยา         3.0  หอรัตนไชย    1.0  สำนักงานสาธารณสุขจังหวัด   
4      เมืองอ่างทอง         8.0    บ้านอิฐ    1.0  สำนักงานสาธารณสุขจังหวัด   

   code9 name10 short_thai short_english  
0    NaN    NaN         สป           SPK  
1    NaN    NaN         นบ           NBI  
2    NaN    NaN         ปท           PTE  
3    NaN    NaN         อย           AYA  
4    NaN    NaN         อท           ATG

In [43]:
hospital = hospital[['hcode','name','zone_name','province','amphur','tambon','name5']]      # เลือกเฉพาะคอลัมน์ที่ต้องการ
print(hospital.shape)
hospital.head()

(17103, 7)


hcode                                     name             zone_name  \
0  00001      สำนักงานสาธารณสุขจังหวัดสมุทรปราการ    สาขาพื้นที่(ระยอง)   
1  00002          สำนักงานสาธารณสุขจังหวัดนนทบุรี  สาขาพื้นที่(สระบุรี)   
2  00003         สำนักงานสาธารณสุขจังหวัดปทุมธานี  สาขาพื้นที่(สระบุรี)   
3  00004  สำนักงานสาธารณสุขจังหวัดพระนครศรีอยุธยา  สาขาพื้นที่(สระบุรี)   
4  00005          สำนักงานสาธารณสุขจังหวัดอ่างทอง  สาขาพื้นที่(สระบุรี)   

          province            amphur     tambon                     name5  
0      สมุทรปราการ  เมืองสมุทรปราการ     ปากน้ำ  สำนักงานสาธารณสุขจังหวัด  
1          นนทบุรี      เมืองนนทบุรี   บางกระสอ  สำนักงานสาธารณสุขจังหวัด  
2         ปทุมธานี     เมืองปทุมธานี    บางปรอก  สำนักงานสาธารณสุขจังหวัด  
3  พระนครศรีอยุธยา   พระนครศรีอยุธยา  หอรัตนไชย  สำนักงานสาธารณสุขจังหวัด  
4          อ่างทอง      เมืองอ่างทอง    บ้านอิฐ  สำนักงานสาธารณสุขจังหวัด

In [44]:
hospital.isnull().sum()     # ดูค่าว่าง

hcode          0
name           1
zone_name      4
province       0
amphur         2
tambon       209
name5          3
dtype: int64

## **C: name**

In [45]:
hospital[hospital['name'].isnull()]

hcode name                  zone_name province      amphur tambon  \
15453  26276  NaN  สาขาพื้นที่(สุราษฎร์ธานี)    ชุมพร  เมืองชุมพร    NaN   

             name5  
15453  คลินิกเอกชน

In [46]:
hospital['name'] = hospital['name'].fillna('ไม่ระบุ')     # แทนค่าว่างของ คอลัมน์ name ด้วย ไม่ระบุ

In [47]:
hospital.isnull().sum()     # ดูค่าว่าง

hcode          0
name           0
zone_name      4
province       0
amphur         2
tambon       209
name5          3
dtype: int64

## **C: zone_name**

In [48]:
set(hospital['zone_name'])

{'08',
 nan,
 'สาขาพื้นที่(ขอนแก่น)',
 'สาขาพื้นที่(นครราชสีมา)',
 'สาขาพื้นที่(นครสวรรค์)',
 'สาขาพื้นที่(พิษณุโลก)',
 'สาขาพื้นที่(ระยอง)',
 'สาขาพื้นที่(ราชบุรี)',
 'สาขาพื้นที่(สงขลา)',
 'สาขาพื้นที่(สระบุรี)',
 'สาขาพื้นที่(สุราษฎร์ธานี)',
 'สาขาพื้นที่(อุดรธานี)',
 'สาขาพื้นที่(อุบลราชธานี)',
 'สาขาพื้นที่(เชียงใหม่)',
 'สาขาพื้นที่กรุงเทพมหานคร'}

In [49]:
hospital[(hospital['zone_name'].isnull()) | (hospital['zone_name']=='08')]      # เลือกดูค่าว่าง และ ค่าผิดปกติ

hcode                                               name zone_name  \
16800  40864                        โรงพยาบาลสรรพสิทธิ์อินเตอร์       NaN   
16801  40886  สำนักงานป้องกันควบคุมโรคที่ 8 จังหวัดอุดรธานี\...        08   
16803  41092            โรงพยาบาลส่งเสริมสุขภาพตำบลหนองจานพัฒนา       NaN   
16805  77609                          หน่วยบริการปฐมภูมิหัวทุ่ง       NaN   
16810  77651                               โรงพยาบาลโคกโพธิ์ไชย       NaN   

          province            amphur      tambon                        name5  
16800  อุบลราชธานี  เมืองอุบลราชธานี     ในเมือง               โรงพยาบาลเอกชน  
16801     อุดรธานี     เมืองอุดรธานี         NaN         ศูนย์บริการสาธารณสุข  
16803   นครราชสีมา        หนองบุญมาก  หนองบุนนาก  โรงพยาบาลส่งเสริมสุขภาพตำบล  
16805      ขอนแก่น      เมืองขอนแก่น    บ้านเป็ด           หน่วยบริการปฐมภูมิ  
16810      ขอนแก่น      เมืองขอนแก่น     บ้านโคก              โรงพยาบาลทั่วไป

In [50]:
hospital['zone_name'][16800] = 'สาขาพื้นที่(อุบลราชธานี)'     # แทนค่าว่าง ด้วยค่าที่มีความเป็นไปได้
hospital['zone_name'][16801] = 'สาขาพื้นที่(อุดรธานี)'
hospital['zone_name'][16803] = 'สาขาพื้นที่(นครราชสีมา)'
hospital['zone_name'][16805] = 'สาขาพื้นที่(ขอนแก่น)'
hospital['zone_name'][16810] = 'สาขาพื้นที่(ขอนแก่น)'

In [51]:
set(hospital['zone_name'])

{'สาขาพื้นที่(ขอนแก่น)',
 'สาขาพื้นที่(นครราชสีมา)',
 'สาขาพื้นที่(นครสวรรค์)',
 'สาขาพื้นที่(พิษณุโลก)',
 'สาขาพื้นที่(ระยอง)',
 'สาขาพื้นที่(ราชบุรี)',
 'สาขาพื้นที่(สงขลา)',
 'สาขาพื้นที่(สระบุรี)',
 'สาขาพื้นที่(สุราษฎร์ธานี)',
 'สาขาพื้นที่(อุดรธานี)',
 'สาขาพื้นที่(อุบลราชธานี)',
 'สาขาพื้นที่(เชียงใหม่)',
 'สาขาพื้นที่กรุงเทพมหานคร'}

In [52]:
hospital.isnull().sum()

hcode          0
name           0
zone_name      0
province       0
amphur         2
tambon       209
name5          3
dtype: int64

## **C: amphur**

In [53]:
hospital[hospital['amphur'].isnull()]

hcode                name                  zone_name       province  \
15064  25733  สารีคลินิกการแพทย์   สาขาพื้นที่กรุงเทพมหานคร  กรุงเทพมหานคร   
15180  25852    คลินิกแพทย์สุเชต  สาขาพื้นที่(สุราษฎร์ธานี)         ภูเก็ต   

      amphur tambon        name5  
15064    NaN    NaN  คลินิกเอกชน  
15180    NaN    NaN  คลินิกเอกชน

- จากการ serch หาข้อมูล พบว่า คลินิกแพทย์สุเชต ตั้งอยู่ที่ อำเภอถลาง จังหวัดภูเก็ต



In [54]:
hospital['amphur'][15064] = 'ไม่ระบุ'     # แทนค่าว่าง ด้วยค่าที่มีความเป็นไปได้
hospital['amphur'][15180] = 'ถลาง'

In [55]:
hospital.isnull().sum()

hcode          0
name           0
zone_name      0
province       0
amphur         0
tambon       209
name5          3
dtype: int64

## **C: tambon**

In [56]:
hospital['tambon'] = hospital['tambon'].fillna('ไม่ระบุ')     # แทนค่าว่างของ ตำบล ด้วย ไม่ระบุ

In [57]:
hospital.isnull().sum()

hcode        0
name         0
zone_name    0
province     0
amphur       0
tambon       0
name5        3
dtype: int64

## **C: name5 (type)**

In [58]:
hospital = hospital.rename(columns={'name5':'type'})  #     เปลี่ยนชื่อ คอลัมน์ name5 เป็น type
print(hospital.shape)
hospital.head()

(17103, 7)


hcode                                     name             zone_name  \
0  00001      สำนักงานสาธารณสุขจังหวัดสมุทรปราการ    สาขาพื้นที่(ระยอง)   
1  00002          สำนักงานสาธารณสุขจังหวัดนนทบุรี  สาขาพื้นที่(สระบุรี)   
2  00003         สำนักงานสาธารณสุขจังหวัดปทุมธานี  สาขาพื้นที่(สระบุรี)   
3  00004  สำนักงานสาธารณสุขจังหวัดพระนครศรีอยุธยา  สาขาพื้นที่(สระบุรี)   
4  00005          สำนักงานสาธารณสุขจังหวัดอ่างทอง  สาขาพื้นที่(สระบุรี)   

          province            amphur     tambon                      type  
0      สมุทรปราการ  เมืองสมุทรปราการ     ปากน้ำ  สำนักงานสาธารณสุขจังหวัด  
1          นนทบุรี      เมืองนนทบุรี   บางกระสอ  สำนักงานสาธารณสุขจังหวัด  
2         ปทุมธานี     เมืองปทุมธานี    บางปรอก  สำนักงานสาธารณสุขจังหวัด  
3  พระนครศรีอยุธยา   พระนครศรีอยุธยา  หอรัตนไชย  สำนักงานสาธารณสุขจังหวัด  
4          อ่างทอง      เมืองอ่างทอง    บ้านอิฐ  สำนักงานสาธารณสุขจังหวัด

In [59]:
set(hospital['type'])

{nan,
 'คลินิกเอกชน',
 'ศูนย์บริการสาธารณสุข',
 'ศูนย์วิชาการ',
 'ศูนย์สุขภาพชุมชน',
 'ศูนย์สุขภาพชุมชน ของ รพ.',
 'สถานบริการสาธารณสุขชุมชน',
 'สถานีอนามัย',
 'สำนักงานสาธารณสุขจังหวัด',
 'สำนักงานสาธารณสุขอำเภอ/กิ่งอำเภอ',
 'หน่วยบริการปฐมภูมิ',
 'โรงพยาบาล นอก สธ.',
 'โรงพยาบาล นอก สป.สธ.',
 'โรงพยาบาล/ศูนย์บริการสาธารณสุข สาขา',
 'โรงพยาบาลชุมชน',
 'โรงพยาบาลทั่วไป',
 'โรงพยาบาลศูนย์',
 'โรงพยาบาลส่งเสริมสุขภาพตำบล',
 'โรงพยาบาลส่งเสริมสุขภาพตำบล ',
 'โรงพยาบาลเอกชน'}

In [60]:
hospital[hospital['type'].isnull()]

hcode                 name             zone_name province       amphur  \
16474  28268  คลินิกนายแพทย์สันติ  สาขาพื้นที่(สระบุรี)   ลพบุรี  เมืองลพบุรี   
16484  28278         คลินิกหมอปาน  สาขาพื้นที่(สระบุรี)   ลพบุรี  เมืองลพบุรี   
16486  28280     คลินิกหมอวิภาวดี  สาขาพื้นที่(สระบุรี)   ลพบุรี  เมืองลพบุรี   

          tambon type  
16474  ทะเลชุบศร  NaN  
16484       กกโก  NaN  
16486       กกโก  NaN

In [61]:
hospital['type'] = hospital['type'].fillna('คลินิกเอกชน')     # แทนค่าว่างของ type ด้วยค่าที่มีความเป็นไปได้

In [62]:
hospital['type'].replace({'โรงพยาบาลส่งเสริมสุขภาพตำบล ':'โรงพยาบาลส่งเสริมสุขภาพตำบล'},inplace=True)   #เปลี่ยนข้อมูลที่เขียนผิด

In [63]:
set(hospital['type'])

{'คลินิกเอกชน',
 'ศูนย์บริการสาธารณสุข',
 'ศูนย์วิชาการ',
 'ศูนย์สุขภาพชุมชน',
 'ศูนย์สุขภาพชุมชน ของ รพ.',
 'สถานบริการสาธารณสุขชุมชน',
 'สถานีอนามัย',
 'สำนักงานสาธารณสุขจังหวัด',
 'สำนักงานสาธารณสุขอำเภอ/กิ่งอำเภอ',
 'หน่วยบริการปฐมภูมิ',
 'โรงพยาบาล นอก สธ.',
 'โรงพยาบาล นอก สป.สธ.',
 'โรงพยาบาล/ศูนย์บริการสาธารณสุข สาขา',
 'โรงพยาบาลชุมชน',
 'โรงพยาบาลทั่วไป',
 'โรงพยาบาลศูนย์',
 'โรงพยาบาลส่งเสริมสุขภาพตำบล',
 'โรงพยาบาลเอกชน'}

In [64]:
hospital.isnull().sum()

hcode        0
name         0
zone_name    0
province     0
amphur       0
tambon       0
type         0
dtype: int64

In [65]:
print(hospital.shape)       # ตาราง hospital ที่ clean แล้ว
hospital.head()

(17103, 7)


hcode                                     name             zone_name  \
0  00001      สำนักงานสาธารณสุขจังหวัดสมุทรปราการ    สาขาพื้นที่(ระยอง)   
1  00002          สำนักงานสาธารณสุขจังหวัดนนทบุรี  สาขาพื้นที่(สระบุรี)   
2  00003         สำนักงานสาธารณสุขจังหวัดปทุมธานี  สาขาพื้นที่(สระบุรี)   
3  00004  สำนักงานสาธารณสุขจังหวัดพระนครศรีอยุธยา  สาขาพื้นที่(สระบุรี)   
4  00005          สำนักงานสาธารณสุขจังหวัดอ่างทอง  สาขาพื้นที่(สระบุรี)   

          province            amphur     tambon                      type  
0      สมุทรปราการ  เมืองสมุทรปราการ     ปากน้ำ  สำนักงานสาธารณสุขจังหวัด  
1          นนทบุรี      เมืองนนทบุรี   บางกระสอ  สำนักงานสาธารณสุขจังหวัด  
2         ปทุมธานี     เมืองปทุมธานี    บางปรอก  สำนักงานสาธารณสุขจังหวัด  
3  พระนครศรีอยุธยา   พระนครศรีอยุธยา  หอรัตนไชย  สำนักงานสาธารณสุขจังหวัด  
4          อ่างทอง      เมืองอ่างทอง    บ้านอิฐ  สำนักงานสาธารณสุขจังหวัด

# **รวมตาราง**

In [66]:
all_data = alldb.merge(hospital,how='left',left_on='site',right_on='hcode')     # เชื่อมตาราง
print(all_data.shape)
all_data.head()

(38171, 18)


user                                RowKey       date ov  \
0  06223                        kfpwuaqldj.png 2022-03-23  0   
1  12015  00859DB9-1BD9-4825-B246-3DC25EFD1924 2022-03-30  4   
2  91811  1376ACA2-AFA5-4457-967A-D2FCF908E6A5 2022-03-30  4   
3  95821  1914EE67-CD8F-47EE-A968-2772911AA5DF 2022-03-30  3   
4  13361  1A219EE7-3647-4D7D-A8C0-AE638E69D889 2022-03-29  3   

               phone   site      time            urine PartitionKey      sex  \
0  Xiaomi 21081111RG  13777  16:31:45  ปกติ (เหลืองใส)        00000  ไม่ระบุ   
1         iPhone12,3  06307  14:46:08  ปกติ (เหลืองใส)        00001     หญิง   
2         iPhone14,3  05328  15:27:45  ปกติ (เหลืองใส)        00001     หญิง   
3         iPhone13,4  06302  10:16:46  ปกติ (เหลืองใส)        00001     หญิง   
4          iPhone9,4  05059  12:26:38  ปกติ (เหลืองใส)        00001      ชาย   

   age  hcode                                               name  \
0   54  13777            โรงพยาบาลศรีนครินทร์ มหาวิทยาลัยขอนแก่น   
1   47  06307  โรงพยาบาลส่งเสริมสุขภาพตำบลบ้านเกาะ  ตำบลบ้านดารา   
2   41  05328        โรงพยาบาลส่งเสริมสุขภาพตำบลบ้านโคกประสิทธิ์   
3   55  06302                 โรงพยาบาลส่งเสริมสุขภาพตำบลสองห้อง   
4   66  05059              โรงพยาบาลส่งเสริมสุขภาพตำบลบ้านดอนดู่   

               zone_name   province        amphur    tambon  \
0   สาขาพื้นที่(ขอนแก่น)    ขอนแก่น  เมืองขอนแก่น   ในเมือง   
1  สาขาพื้นที่(พิษณุโลก)  อุตรดิตถ์         พิชัย  บ้านดารา   
2   สาขาพื้นที่(ขอนแก่น)  กาฬสินธุ์       ฆ้องชัย  โคกสะอาด   
3  สาขาพื้นที่(พิษณุโลก)  อุตรดิตถ์        ฟากท่า   สองห้อง   
4   สาขาพื้นที่(ขอนแก่น)  มหาสารคาม         นาดูน     หัวดง   

                          type  
0            โรงพยาบาล นอก สธ.  
1  โรงพยาบาลส่งเสริมสุขภาพตำบล  
2  โรงพยาบาลส่งเสริมสุขภาพตำบล  
3  โรงพยาบาลส่งเสริมสุขภาพตำบล  
4  โรงพยาบาลส่งเสริมสุขภาพตำบล

In [67]:
all_data.isnull().sum()

user             0
RowKey           0
date             0
ov               0
phone            0
site             0
time             0
urine            0
PartitionKey     0
sex              0
age              0
hcode           76
name            76
zone_name       76
province        76
amphur          76
tambon          76
type            76
dtype: int64

In [68]:
all_data[all_data['hcode'].isnull()]        # ดูข้อมูลที่เป็นค่าว่าง

user                                RowKey       date ov  \
30     05039  6F6ED841-7574-4C1C-AF10-CC6E2658E718 2022-03-28  0   
790    05353  7A4BD7DC-5BD3-4928-B57F-CE7C86E32DAE 2022-03-19  4   
915    05353  218CB548-E5F1-4F5D-87C5-36EC2256B7CB 2022-03-19  4   
1675   05353  7CDA4D3C-6E12-4CF8-A94C-76E5CE4E9202 2022-03-19  4   
1800   05353  6FC7AE57-7365-4DCC-A656-772CBDDDA882 2022-03-19  0   
...      ...                                   ...        ... ..   
38166  99999  F45E78D3-7D89-47A3-8207-C705377E0E0A 2022-03-11  1   
38167  99999                        lnlohgqckf.png 2022-03-10  0   
38168  99999                        vkdocpmjiv.png 2022-03-10  1   
38169  99999                        vrvyfnvayq.png 2022-03-10  1   
38170  99999                        zuhbtvyrce.png 2022-03-10  1   

                  phone   site      time                 urine PartitionKey  \
30           iPhone11,8  45972  10:36:41                  ขุ่น        00001   
790            iPad13,1  75757  17:22:08                  ขุ่น        00006   
915            iPad13,1  75757  17:20:05       ปกติ (เหลืองใส)        00007   
1675           iPad13,1  75757  17:14:16       ปกติ (เหลืองใส)        00012   
1800           iPad13,1  75757  17:15:16                  ขุ่น        00013   
...                 ...    ...       ...                   ...          ...   
38166        iPhone14,2  99999  11:36:34                  ขุ่น        99999   
38167  Samsung SM-G998B  99999  18:10:29               มีตะกอน        99999   
38168  Samsung SM-G998B  99999  12:14:41  ขุ่น, มีเม็ดเลือดแดง        99999   
38169  Samsung SM-G998B  99999  13:58:01       ปกติ (เหลืองใส)        99999   
38170  Samsung SM-G998B  99999  13:57:58       ปกติ (เหลืองใส)        99999   

           sex  age hcode name zone_name province amphur tambon type  
30     ไม่ระบุ   54   NaN  NaN       NaN      NaN    NaN    NaN  NaN  
790    ไม่ระบุ   54   NaN  NaN       NaN      NaN    NaN    NaN  NaN  
915    ไม่ระบุ   54   NaN  NaN       NaN      NaN    NaN    NaN  NaN  
1675   ไม่ระบุ   54   NaN  NaN       NaN      NaN    NaN    NaN  NaN  
1800   ไม่ระบุ   54   NaN  NaN       NaN      NaN    NaN    NaN  NaN  
...        ...  ...   ...  ...       ...      ...    ...    ...  ...  
38166  ไม่ระบุ   54   NaN  NaN       NaN      NaN    NaN    NaN  NaN  
38167  ไม่ระบุ   54   NaN  NaN       NaN      NaN    NaN    NaN  NaN  
38168  ไม่ระบุ   54   NaN  NaN       NaN      NaN    NaN    NaN  NaN  
38169  ไม่ระบุ   54   NaN  NaN       NaN      NaN    NaN    NaN  NaN  
38170  ไม่ระบุ   54   NaN  NaN       NaN      NaN    NaN    NaN  NaN  

[76 rows x 18 columns]

In [69]:
all_data = all_data.dropna()    # ลบข้อมูลที่เป็นค่าว่างทิ้ง
print(all_data.shape)
all_data.head()

(38095, 18)


user                                RowKey       date ov  \
0  06223                        kfpwuaqldj.png 2022-03-23  0   
1  12015  00859DB9-1BD9-4825-B246-3DC25EFD1924 2022-03-30  4   
2  91811  1376ACA2-AFA5-4457-967A-D2FCF908E6A5 2022-03-30  4   
3  95821  1914EE67-CD8F-47EE-A968-2772911AA5DF 2022-03-30  3   
4  13361  1A219EE7-3647-4D7D-A8C0-AE638E69D889 2022-03-29  3   

               phone   site      time            urine PartitionKey      sex  \
0  Xiaomi 21081111RG  13777  16:31:45  ปกติ (เหลืองใส)        00000  ไม่ระบุ   
1         iPhone12,3  06307  14:46:08  ปกติ (เหลืองใส)        00001     หญิง   
2         iPhone14,3  05328  15:27:45  ปกติ (เหลืองใส)        00001     หญิง   
3         iPhone13,4  06302  10:16:46  ปกติ (เหลืองใส)        00001     หญิง   
4          iPhone9,4  05059  12:26:38  ปกติ (เหลืองใส)        00001      ชาย   

   age  hcode                                               name  \
0   54  13777            โรงพยาบาลศรีนครินทร์ มหาวิทยาลัยขอนแก่น   
1   47  06307  โรงพยาบาลส่งเสริมสุขภาพตำบลบ้านเกาะ  ตำบลบ้านดารา   
2   41  05328        โรงพยาบาลส่งเสริมสุขภาพตำบลบ้านโคกประสิทธิ์   
3   55  06302                 โรงพยาบาลส่งเสริมสุขภาพตำบลสองห้อง   
4   66  05059              โรงพยาบาลส่งเสริมสุขภาพตำบลบ้านดอนดู่   

               zone_name   province        amphur    tambon  \
0   สาขาพื้นที่(ขอนแก่น)    ขอนแก่น  เมืองขอนแก่น   ในเมือง   
1  สาขาพื้นที่(พิษณุโลก)  อุตรดิตถ์         พิชัย  บ้านดารา   
2   สาขาพื้นที่(ขอนแก่น)  กาฬสินธุ์       ฆ้องชัย  โคกสะอาด   
3  สาขาพื้นที่(พิษณุโลก)  อุตรดิตถ์        ฟากท่า   สองห้อง   
4   สาขาพื้นที่(ขอนแก่น)  มหาสารคาม         นาดูน     หัวดง   

                          type  
0            โรงพยาบาล นอก สธ.  
1  โรงพยาบาลส่งเสริมสุขภาพตำบล  
2  โรงพยาบาลส่งเสริมสุขภาพตำบล  
3  โรงพยาบาลส่งเสริมสุขภาพตำบล  
4  โรงพยาบาลส่งเสริมสุขภาพตำบล

In [70]:
all_data.isnull().sum()     # ดูค่าว่าง

user            0
RowKey          0
date            0
ov              0
phone           0
site            0
time            0
urine           0
PartitionKey    0
sex             0
age             0
hcode           0
name            0
zone_name       0
province        0
amphur          0
tambon          0
type            0
dtype: int64

In [71]:
print(f'เสียข้อมูลไป {40035-38095} แถว')
print(f'คิดเป็น {((40035-38095)*100)/40035} %')

เสียข้อมูลไป 1940 แถว
คิดเป็น 4.845759960034969 %


**เสียข้อมูลไป 1940 แถว**

**คิดเป็น 4.845759960034969 %**

In [72]:
print(all_data.shape)   # ตารางข้อมูลพร้อมใช้งาน
all_data.head()

(38095, 18)


user                                RowKey       date ov  \
0  06223                        kfpwuaqldj.png 2022-03-23  0   
1  12015  00859DB9-1BD9-4825-B246-3DC25EFD1924 2022-03-30  4   
2  91811  1376ACA2-AFA5-4457-967A-D2FCF908E6A5 2022-03-30  4   
3  95821  1914EE67-CD8F-47EE-A968-2772911AA5DF 2022-03-30  3   
4  13361  1A219EE7-3647-4D7D-A8C0-AE638E69D889 2022-03-29  3   

               phone   site      time            urine PartitionKey      sex  \
0  Xiaomi 21081111RG  13777  16:31:45  ปกติ (เหลืองใส)        00000  ไม่ระบุ   
1         iPhone12,3  06307  14:46:08  ปกติ (เหลืองใส)        00001     หญิง   
2         iPhone14,3  05328  15:27:45  ปกติ (เหลืองใส)        00001     หญิง   
3         iPhone13,4  06302  10:16:46  ปกติ (เหลืองใส)        00001     หญิง   
4          iPhone9,4  05059  12:26:38  ปกติ (เหลืองใส)        00001      ชาย   

   age  hcode                                               name  \
0   54  13777            โรงพยาบาลศรีนครินทร์ มหาวิทยาลัยขอนแก่น   
1   47  06307  โรงพยาบาลส่งเสริมสุขภาพตำบลบ้านเกาะ  ตำบลบ้านดารา   
2   41  05328        โรงพยาบาลส่งเสริมสุขภาพตำบลบ้านโคกประสิทธิ์   
3   55  06302                 โรงพยาบาลส่งเสริมสุขภาพตำบลสองห้อง   
4   66  05059              โรงพยาบาลส่งเสริมสุขภาพตำบลบ้านดอนดู่   

               zone_name   province        amphur    tambon  \
0   สาขาพื้นที่(ขอนแก่น)    ขอนแก่น  เมืองขอนแก่น   ในเมือง   
1  สาขาพื้นที่(พิษณุโลก)  อุตรดิตถ์         พิชัย  บ้านดารา   
2   สาขาพื้นที่(ขอนแก่น)  กาฬสินธุ์       ฆ้องชัย  โคกสะอาด   
3  สาขาพื้นที่(พิษณุโลก)  อุตรดิตถ์        ฟากท่า   สองห้อง   
4   สาขาพื้นที่(ขอนแก่น)  มหาสารคาม         นาดูน     หัวดง   

                          type  
0            โรงพยาบาล นอก สธ.  
1  โรงพยาบาลส่งเสริมสุขภาพตำบล  
2  โรงพยาบาลส่งเสริมสุขภาพตำบล  
3  โรงพยาบาลส่งเสริมสุขภาพตำบล  
4  โรงพยาบาลส่งเสริมสุขภาพตำบล

In [73]:
# all_data.to_csv('all_data.csv',index=False)       # download ข้อมูล